manual/package_interface.md

```diff
@@ -0,0 +1,624 @@
+| [package:has_cxxfuncs](#packagehas_cxxfuncs)           | Wether the package has the given C++ functions                               | >= 2.2.5           |
+| [package:has_ctypes](#packagehas_ctypes)               | Wether the package has the given C types                                     | >= 2.3.8           |
+| [package:has_cxxtypes](#packagehas_cxxtypes)           | Wether the package has the given C++ types                                   | >= 2.3.8           |
+| [package:has_cincludes](#packagehas_cincludes)         | Wether the package has the given C header files                              | >= 2.3.8           |
+| [package:has_cxxincludes](#packagehas_cxxincludes)     | Wether the package has the given C++ header files                            | >= 2.3.8           |
+| [package:check_csnippets](#packagecheck_csnippets)     | Wether the given C snippet can be compiled and linked                        | >= 2.2.6           |
+| [package:check_cxxsnippets](#packagecheck_cxxsnippets) | Wether the given C++ snippet can be compiled and linked                      | >= 2.2.6           |
+
+#### package:name
+
+- Get the name of the package
+
+#### package:get
+
+- Get the values of the package by name
+
+```lua
+-- get the dependencies
+package:get("deps")
+-- get the links
+package:get("links")
+-- get the defined macros
+package:get("defines")
+```
+
+#### package:set
+
+- Set the values of the package by name (If you just want to add values use [package:add](#packageadd))
+
+```lua
+-- set the dependencies
+package:set("deps", {"python"})
+-- set the links
+package:set("links", {"sdl2"})
+-- set the defined macros
+package:set("defines", {"SDL_MAIN_HANDLED"})
+```
+
+#### package:add
+
+- Add to the values of the package by name
+
+```lua
+-- add dependencies
+package:add("deps", "python")
+-- add links
+package:add("links", "sdl2")
+-- add defined macros
+package:add("defines", "SDL_MAIN_HANDLED")
+```
+
+#### package:license
+
+- Get the license of the package (Same as `package:get("license")`)
+
+#### package:description
+
+- Get the description of the package (Same as `package:get("description")`)
+
+#### package:plat
+
+- Get the platform of the package. Can be any of:
+  + windows
+  + linux
+  + macosx
+  + android
+  + iphoneos
+  + watchos
+  + mingw
+  + cygwin
+  + bsd
+
+If the package is binary [`os.host`](manual/builtin_modules.md#oshost) is returned
+
+#### package:arch
+
+- Get the architecture of the package (e.g. x86, x64, x86_64)
+
+If the package is binary [`os.arch`](manual/builtin_modules.md#osarch) is returned
+
+#### package:targetos
+
+- Get the targeted OS of the package. Can have the same values as [package:plat](#packageplat)
+
+#### package:targetarch
+
+- Get the targeted architecture of the package. Can have the same values as [package:arch](#packagearch)
+
+#### package:mode
```

Comment:
It is private api now, please remove it. We need only `package:is_debug()` and only support release/debug mode now.